reviewの平均の表記に関して #75

mamiko-7172 opened the pull request:
〇indexの平均を表す部分のコードを修正
　review→ratingに変更
〇Storecontrollerのshowメソッドに追記
　

mamiko-7172 committed:
reviewの平均の表記に関して

〇indexの平均を表す部分のコードを修正
　review→ratingに変更
〇Storecontrollerのshowメソッドに追記
　

chihiro0415 commented on app/Http/Controllers/StoreController.php:
```diff
@@ -65,7 +65,21 @@ public function store(Request $request)
     {
 
-        $store = Store::find($id);
+        $stores = Store::where('$id','=',$id);
+
+        foreach($stores as $store) {
+            $ratings = $store->review_ratings();
+
+            if(count($ratings) < 1) continue;
+            
+            $sum = 0;
+            foreach ($ratings as  $id=>$value) {
+                $sum+=$value;
+            }
+            $rating = $sum / count($ratings);
+            $store -> rating = $rating;
+        }
+
```
お店の詳細ページ、なのでお店は複数ではなく1つですね！

なので
```
        $store = Store::find($id);
        $ratings = $store->review_ratings();
        
        $sum = 0;
        foreach ($ratings as  $id=>$value) {
            $sum+=$value;
        }
        $rating = $sum / count($ratings);
        $store -> rating = $rating;
```

こちらのみでよいかと〜！
（ただこれだと小数点いっぱいでちゃうので、round関数を使って小数点第一位までの表示にしても良さそうです！）

mamiko-7172 replied:
@chihiro0415 
夜分遅くにコメントありがとうございます。
コメントいただいた通りに修正してみたのですが、「Division by zero」のエラーが出てしまったので
if文を入れてみたのですがどうでしょうか？

chihiro0415 replied:
確かにif文いいですね！！レビューがない場合はifで0にしてあげる必要がありますね:+1:

とすると、こんな感じが良いかと思います！（ちょっと修正しました）
```
    public function show($id)
    {
        $store = Store::find($id);
        $ratings = $store->review_ratings();

        $sum = 0;
        foreach ($ratings as  $id=>$value) {
            $sum+=$value;
        }

        if($sum == 0){
            $store -> rating = 0;
        }else{
            $store -> rating = round($sum / count($ratings), 1);
        }

        return view ('stores.show', ['store'=>$store]);
    }
```

mamiko-7172 committed:
コメントをいただいた内容をもとに修正

修正したところ「Division by zero」のエラーが出てしまったので
if文でエラーが出ないようにした方がいいのかと思ったのですが
どうなんでしょうか？？

mamiko-7172 committed:
式を変更しました

mamiko-7172 committed:
頂いたコメント通り修正しました！
デバック確認でもデータは飛ばせていたので大丈夫かと思います。

chihiro0415 commented:
すみません、ついでに今回対応したshowメソッドの部分、view側でも評価が表示されるように修正してもらえませんか..？:pray:

show.blade.php（36~42行目あたり）
**現状**
```
                    @foreach ($store->reviews as $review)
                    <div v-for="star in [5,4,3,2,1]">
                        <input v-model="reviewParams.stars" type="radio" :value="star">
                        <v-star :value="star">{{$review->review}}</v-star>
                    </div>
                    <span class="value">{{$review->review}}</span>
                    @endforeach
```

ここがまるっと↓
**理想**
```
<span class="value">{{$store->rating}}</span>
```
これのみで大丈夫です！
（デザイン通りでいくなら星も表示したいところですが、せめて数字だけでも。。。！）

mamiko-7172 committed:
show.bladeの口コミを表示すうためのコードの変更

show.bladeの154～193行目を変更しました
indexで表示させた時と同じ考えでやってみたのですがどうでしょうか。

chihiro0415 commented on resources/views/stores/show.blade.php:
```diff
@@ -151,43 +151,43 @@
-                        <dd>{{$review->ate_thing}}</dd>
+                        <dd>{{$review->reviews()['ate_thing']}}</dd>
                         <dt class="post-title">カテゴリー：</dt>
-                        <dd>{{$review->category_id}}</dd>
+                        <dd>{{$review->reviews()['category_id']}}</dd>
                         <dt class="post-title">支払額：</dt>
-                        <dd>{{$review->charge}}円</dd>
+                        <dd>{{$review->reviews()['charge']}}円</dd>
                     </dl>
                     <div class="average">
                         <span class="star">
                             <span>
-                                {{$review->review > 1 ? '★' : '☆' }}
+                                {{$review->reviews()['review'] >= 1 ? '★' : '☆' }}
                             </span>
                             <span>
-                                {{$review->review > 2 ? '★' : '☆' }}
+                                {{$review->reviews()['review'] >= 2 ? '★' : '☆' }}
                             </span>
                             <span>
-                                {{$review->review > 3 ? '★' : '☆' }}
+                                {{$review->reviews()['review'] >= 3 ? '★' : '☆' }}
                             </span>
                             <span>
-                                {{$review->review > 4 ? '★' : '☆' }}
+                                {{$review->reviews()['review'] >= 4 ? '★' : '☆' }}
                             </span>
                             <span>
-                                {{$review->review > 5 ? '★' : '☆' }}
+                                {{$review->reviews()['review'] >= 5 ? '★' : '☆' }}
                             </span>
                         </span>
-                        <span class="value">{{$review->review}}</span>
+                        <span class="value">{{$review->reviews()['review']}}</span>
                     </div>
                 </div>
-                @if ($review->post_image !== null)
-                <img src="{{$review->post_image}}" alt="投稿写真">
+                @if ($review->reviews()['post_image'] !== null)
+                <img src="{{$review->reviews()['post_image']}}" alt="投稿写真">
```
このあたりは何もかえずで大丈夫ですー！

mamiko-7172 replied:
変えなくて大丈夫だったんですね^^;

chihiro0415 replied:
今の最新のmasterで諸々修正済みなので！
確認ありがとうございます！

mamiko-7172 replied:
このコミットを打ち消すようにSauceTreeで打ってみたのですが、それで消えますか？？

mamiko-7172 committed:
show.bladeの店舗の評価部分の星を表示させることと平均値を出すためのコードを追加

mamiko-7172 committed:
Merge branch 'master' into reveiw_ave

Mkk818 commented on resources/views/index.blade.php:
```diff
@@ -334,22 +334,22 @@
                     </span>
                 </span>
-                <span class="value font">{{$detail->review}}</span>
+                <span class="value font">{{$detail->rating}}</span>
```
この部分0の場合でも0と表示されるように変更できますでしょうか？
showのお店の評価平均はLGTMです！

mamiko-7172 replied:
承知しました！

mamiko-7172 committed:
indexでも評価が0だったときに0評価が出るように設定

mamiko-7172 committed:
Merge branch 'reveiw_ave' of https://github.com/ArisaNAKAMURA-GitHub/sweetsguide into reveiw_ave

tai-namechan reviewed (approved):
お店の平均でました！LGTM
